### DISCUSSION TITLE
Accessing files in the /public folder server side on vercel

### DISCUSSION THREAD

**Discussion by krisvegh** [vercel/next.js#44467] [Help] [answered]
Hi,

I'm having issues accessing the files in the `public` folder on the server side when the app is deployed on Vercel. Consider the following folder structure

```
└── public/
    └── 90x50/
        ├── pc_001.jpg
        └── pc_002.jpg
└── server/
    └── renderer/
        └── template-renderer.ts
```

I need to have `pc_001.jpg` and `pc_002.jpg` files to be in the public folder, as I need to access them on the client side. But I also want to use them on the server side, as the backend is generating a PDFs using the same images. So basically I'd like to access these files from inside the `template-renderer.ts`. 

Running the server on `local` that is no problem as a simply 
```ts
fs.readFileSync('public/50x90/pc_001.ts')
```
works fine.. but once deployed on vercel, this path fails `Error: ENOENT: no such file or directory`. 

Based on the `output` menu in vercel, it seems like the `public` folder is being removed, and all it's contents are moved to the app root:

<img width="784" alt="image" src="https://user-images.githubusercontent.com/14986242/210140733-f58e586a-2cb3-40e6-991a-4d2060ebb56d.png">

I've also tried accessing `__dirname` in the `next.config` file and also tried  using `path.resolve('./public', '50x90', 'pc_002.jpg')` ([described here](https://medium.com/@boris.poehland.business/next-js-api-routes-how-to-read-files-from-directory-compatible-with-vercel-5fb5837694b9)) . I've also tried to list the folders in the deployment but I don't seem to find the `50x90` folder with the images at all on the Vercel deployment. I'm out of ideas. 

(I know I could just copy the images to the server folder, but has quite a few images in it and I want to avoid duplicating them.)

Does anyone know how I could access those files?

(Currently using nextJS v.12.0.7)

**Comment by krisvegh** [comment, marked as answer]
If someone stumbles into this with the same issue, I've managed to access the files in the public folder from inside a serverless function using `process.cwd()` in the path:

```ts
const imgPath = path.join(process.cwd(), 'public', '90x50/pc_001jpg');
```

**Reply by medovanx** [reply]
Thank you, this is perfect! it solved my issue.
